Rebaseline

sysrqb opened the pull request:
#### 25f472a6b8b0ef4c14e12d1417995654720ef317
<pre>
Rebaseline

* LayoutTests/imported/w3c/web-platform-tests/FileAPI/url/multi-global-origin-serialization.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/historical.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/historical.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/WebCryptoAPI/historical.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/beacon/headers/header-referrer-no-referrer-when-downgrade.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/beacon/headers/header-referrer-origin-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/beacon/headers/header-referrer-origin-when-cross-origin-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/beacon/headers/header-referrer-same-origin-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/beacon/headers/header-referrer-strict-origin-when-cross-origin.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/beacon/headers/header-referrer-strict-origin.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/beacon/headers/header-referrer-unsafe-url.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/connect-src/connect-src-beacon-redirect-to-blocked.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/connect-src/shared-worker-connect-src-allowed.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/font-src/font-match-allowed.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/form-action/form-action-src-blocked.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/form-action/form-action-src-redirect-blocked.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/frame-ancestors/frame-ancestors-nested-cross-in-same-none-block-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/frame-ancestors/frame-ancestors-nested-cross-in-same-self-block-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/frame-ancestors/frame-ancestors-nested-cross-in-same-star-allow-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/frame-ancestors/frame-ancestors-nested-cross-in-same-url-allow-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/frame-ancestors/frame-ancestors-nested-cross-in-same-url-block-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/frame-ancestors/frame-ancestors-self-block-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/frame-ancestors/frame-ancestors-star-allow-crossorigin-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/frame-ancestors/frame-ancestors-url-block-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/frame-src/frame-src-same-document-meta.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/frame-src/frame-src-same-document.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/script-src-wildcard/script-tag.http-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/script-src-wildcard/sharedworker-import.http-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/script-src-wildcard/sharedworker-import.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/script-src-wildcard/worklet-audio.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-none/script-tag.http-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-none/script-tag.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-none/worklet-audio-import-data.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-none/worklet-audio.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-self/script-tag.http-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-self/script-tag.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-self/sharedworker-import.http-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-self/sharedworker-import.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-self/worklet-audio-import-data.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-self/worklet-audio.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-wildcard/script-tag.http-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-wildcard/script-tag.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-wildcard/sharedworker-import.http-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-wildcard/sharedworker-import.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-wildcard/worklet-audio-import-data.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/worker-src-wildcard/worklet-audio.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/script-src-wildcard/script-tag.http-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/script-src-wildcard/script-tag.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/script-src-wildcard/sharedworker-import.http-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/script-src-wildcard/sharedworker-import.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/script-src-wildcard/worklet-audio.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/worker-src-none/script-tag.http-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/worker-src-none/script-tag.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/worker-src-none/worklet-audio-import-data.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/worker-src-none/worklet-audio.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/worker-src-self/script-tag.http-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/worker-src-self/script-tag.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/worker-src-self/worklet-audio-import-data.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/worker-src-self/worklet-audio.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/worker-src-wildcard/script-tag.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/worker-src-wildcard/sharedworker-import.http-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/worker-src-wildcard/sharedworker-import.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/worker-src-wildcard/worklet-audio-import-data.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/worker-src-wildcard/worklet-audio.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/generic/directive-name-case-insensitive.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/generic/generic-0_10.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/generic/generic-0_8.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/content-security-policy/style-src/style-src-injected-stylesheet-allowed.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/animation/offset-path-interpolation-007-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/feature-policy/feature-policy-frame-policy-allowed-for-all.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/feature-policy/feature-policy-frame-policy-allowed-for-self.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/feature-policy/feature-policy-frame-policy-allowed-for-some-override.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/feature-policy/feature-policy-frame-policy-allowed-for-some.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/feature-policy/feature-policy-frame-policy-disallowed-for-all.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/feature-policy/feature-policy-frame-policy-timing.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/feature-policy/feature-policy-header-policy-allowed-for-some.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/feature-policy/feature-policy-header-policy-declined.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/mode-no-cors.sub.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/mode-no-cors.sub.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/mode-same-origin.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/mode-same-origin.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/response-url.sub.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/response-url.sub.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/scheme-blob.sub.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/scheme-blob.sub.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/scheme-others.sub.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/scheme-others.sub.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/scheme-others.sub.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/scheme-others.sub.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/cors/cors-cookies.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/cors/cors-cookies.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/cors/cors-multiple-origins.sub.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/cors/cors-multiple-origins.sub.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/cors/cors-multiple-origins.sub.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/cors/cors-multiple-origins.sub.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/request/request-init-001.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/fetch-preflight.https.sub.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/fetch-preflight.https.sub.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/fetch-preflight.https.sub.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/fetch-preflight.https.sub.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/fetch.https.sub.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/fetch.https.sub.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/fetch.https.sub.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/fetch.https.sub.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/generated/audioworklet.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/generated/css-font-face.https.sub.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/generated/css-font-face.sub.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/generated/element-script.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/generated/fetch-via-serviceworker.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/generated/fetch.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/generated/script-module-import-dynamic.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/generated/script-module-import-static.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/generated/worker-dedicated-constructor.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/generated/worker-dedicated-importscripts.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/metadata/portal.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/orb/tentative/script-unlabeled.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/private-network-access/iframe.tentative.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/security/dangling-markup-mitigation-data-url.tentative.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/hr-time/timeOrigin-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/indexeddb.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/require-corp-embed-anonymous-iframe.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/history-traversal/browsing_context_name_cross_origin-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/child-navigates-parent-cross-origin.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/cross-origin-top-navigation-with-user-activation-in-parent.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/cross-origin-top-navigation-without-user-activation.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/initial-empty-document/iframe-src-204-fragment-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/javascript-url-security-check-failure.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/javascript-url-security-check-multi-globals.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/javascript-url-security-check-same-origin-domain.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/multiple-globals/context-for-location-assign-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/a-click-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/a-user-click-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/location-assign-user-click-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/location-setter-user-click-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/location-setter-user-mouseup-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/overlapping-navigations-and-traversals/cross-document-traversal-cross-document-nav-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/overlapping-navigations-and-traversals/same-document-traversal-same-document-traversal-hashchange-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/history/the-history-interface/history_pushstate_url_rewriting-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/history/the-location-interface/allow_prototype_cycle_through_location.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/history/the-location-interface/location-prototype-setting-cross-origin.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/history/the-location-interface/location-prototype-setting-same-origin-domain.sub-expected.txt:
</pre>
----------------------------------------------------------------------
#### a3a2dd4062f14b8d7a0e949b24d1d4402b14db33
<pre>
Add support for running a local DNS resolver for WPT
<a href="https://bugs.webkit.org/show_bug.cgi?id=261038">https://bugs.webkit.org/show_bug.cgi?id=261038</a>
<a href="https://rdar.apple.com/82927182">rdar://82927182</a>

Reviewed by NOBODY (OOPS!).

Some ports don&apos;t have support for resolving arbitrary domains, but Web Platform
Tests rely on this capability. Currently, we only use localhost and 127.0.0.1
as a workaround where it&apos;s necessary, but this affects our test coverage. This
patch adds a very basic DNS resolver and introduces a new command-line argument
for run-webkit-tests that enables and uses this resolver on Apple ports.

In 254273@main, glib ports gained support for resolving arbitrary domains, and
this moves Apple ports closer to supporting that.

Previous attempts were made in <a href="https://bugs.webkit.org/show_bug.cgi?id=245294">https://bugs.webkit.org/show_bug.cgi?id=245294</a>,
and 264076@main introduced part of that behavior for a different reason. In
some situations, name resolutions falls back on that SPI, as well.

The basic DNS resolver only resolves A records for names ending with
&quot;localhost&quot; and for the host names defines by the port&apos;s localhost_aliases.
The server always returns 127.0.0.1 as the answer for those hosts. All other
queries return NXDomain. This resolver depends on the dnslib python library.

On Apple ports, the iOS simulator and Mac use different implementations. The
iOS simulator part is in the Network process because we need to define the
resolver in-process otherwise it won&apos;t be used. The way the resolver is
configured on Mac isn&apos;t supported on the simulator. On Mac, we use SPI and a
Network Extensions policy so this works when a VPN is enabled. We do this in
WebKitTestRunner so we don&apos;t need to add additional entitlements and weaken the
sandbox for the network process.

* Source/WebCore/PAL/pal/spi/cocoa/NetworkSPI.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkProcessCocoa.mm:
(WebKit::NetworkProcess::platformInitializeNetworkProcessCocoa):
* Tools/Scripts/webkitpy/__init__.py:
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(parse_args):
* Tools/Scripts/webkitpy/layout_tests/servers/basic_dns_server.py: Added.
(Resolver):
(Resolver.__init__):
(Resolver.resolve):
* Tools/Scripts/webkitpy/layout_tests/servers/web_platform_test_server.py:
(WebPlatformTestServer.__init__):
(WebPlatformTestServer._spawn_process):
(WebPlatformTestServer._stop_running_server):
* Tools/Scripts/webkitpy/port/driver.py:
(Driver.cmd_line):
* Tools/WebKitTestRunner/Configurations/WebKitTestRunner-internal.entitlements:
* Tools/WebKitTestRunner/Options.cpp:
(WTR::handleOptionLocalDNSResolver):
(WTR::OptionsHandler::OptionsHandler):
* Tools/WebKitTestRunner/Options.h:
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
(WTR::TestController::cocoaPlatformInitialize):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8b875fd80398331397ad9cde84160aeb34b758c7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118125 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37803 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28793 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124280 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70164 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5368ce8b-3bc2-4b61-ad4e-f0e5ba19f0c2) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120003 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38496 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46385 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89639 "Found 1 new test failure: imported/w3c/web-platform-tests/webxr/xrWebGLLayer_viewports.https.html (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59269 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c37e8c12-090b-4617-a85f-381137ba6e1d) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121077 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30660 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106220 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70132 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c54496be-b791-4056-bbd2-60c5d3da2b20) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/117485 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29729 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/24331 "Build was cancelled. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Skipped layout-tests; layout-tests (exception)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67947 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100086 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24520 "233 new passes 11 failures") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127357 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45028 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33936 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98317 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45389 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102439 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98103 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43499 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21769 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41473 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44900 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50574 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44360 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47705 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46049 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
```diff
@@ -152,6 +199,45 @@ void initializeWebViewConfiguration(const char* libraryPath, WKStringRef injecte
+    if (!options.useLocalDNSResolver)
+        return;
+
+    NSUUID *agentUUID = [[NSUUID alloc] init];
```
`auto agentUUID = adoptNS([[NSUUID alloc] init]);`

Otherwise, you are leaking.

cdumez commented on Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
```diff
@@ -152,6 +199,45 @@ void initializeWebViewConfiguration(const char* libraryPath, WKStringRef injecte
+        return;
+
+    NSUUID *agentUUID = [[NSUUID alloc] init];
+    if (!agentUUID) {
```
Can this really happen? The document doesn't mention it.

sysrqb replied:
Maybe not. I see similar checks in other code bases, but it's not consistent - and existing usage in WebKit doesn't check the result. I'll remove it.

cdumez commented on Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
```diff
@@ -152,6 +199,45 @@ void initializeWebViewConfiguration(const char* libraryPath, WKStringRef injecte
+    }
+
+    // The following NetworkExtension policy is needed so we can run tests while a VPN is connected
+    NEPolicySession *policySession = [[NEPolicySession alloc] init];
```
Same comment about adoptNS() and a few times below as well.

JonWBedard commented on Tools/Scripts/webkitpy/layout_tests/servers/basic_dns_server.py:
```diff
@@ -0,0 +1,51 @@
+    import time
+    server = DNSServer(Resolver(["site.example"]), port=8053, address="127.0.0.1")
+    server.start_thread()
+    try:
```
I'm a bit confused by this being threaded. Looks like the motivation here is to  be a stand-alone process (which I think is the right design!), I don't understand why we need the spin-loop here.

sysrqb replied:
This was/is only intended to be used for testing as a stand-alone process. When the service is run for WPT, it is only run as a background thread. The spin-loop just allows for using ^C and cleaning-up afterward when testing this script.

sysrqb replied:
I guess you might be asking why I chose this design instead of just calling `server.start()` (https://github.com/paulc/dnslib/blob/master/dnslib/server.py#L384). And I don't know/remember the answer to that - probably for mirroring the usage in the WPT code. I don't have a strong opinion here.

JonWBedard replied:
If it's just for testing, mirroring the usage in WPT is correct. Probably worth noting that this is just intended for testing, though.

gsnedders replied:
> This was/is only intended to be used for testing as a stand-alone process. When the service is run for WPT, it is only run as a background thread. The spin-loop just allows for using ^C and cleaning-up afterward when testing this script.

Just waiting on `Thread.join()` would work here, as ^C will still cause a `KeyboardInterrupt` exception to be thrown.

cdumez commented on Source/WebKit/NetworkProcess/cocoa/NetworkProcessCocoa.mm:
```diff
@@ -111,6 +132,16 @@ static void initializeNetworkSettings()
+
+#if PLATFORM(IOS_FAMILY_SIMULATOR)
+    // See TestController::cocoaPlatformInitialize for supporting a local DNS resolver on Mac.
+    static nw_resolver_config_t resolverConfig = nw_resolver_config_create();
```
Does this need to be static? Should only be called once by process. Or do we need to keep the config alive for the lifetime of the process?

cdumez commented on Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
```diff
@@ -63,6 +64,52 @@
 #import <wtf/spi/cocoa/SecuritySPI.h>
 
+#if PLATFORM(MAC)
+#if USE(APPLE_INTERNAL_SDK)
```
This should really all go in NetworkSPI.h

cdumez commented on Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
```diff
@@ -152,6 +199,40 @@ void initializeWebViewConfiguration(const char* libraryPath, WKStringRef injecte
+    uuid_t agentUUIDBytes;
+    [agentUUID getUUIDBytes:agentUUIDBytes];
+
+    static nw_resolver_config_t resolverConfig = nw_resolver_config_create();
```
Does this really need to be static?

gsnedders commented on Tools/Scripts/webkitpy/layout_tests/servers/basic_dns_server.py:
```diff
@@ -0,0 +1,51 @@
+            time.sleep(1)
+    except Exception:
+        pass
+    server.stop()
```
We probably want this in a `finally` block, to always stop the server, even if an exception like `KeyboardInterrupt` is thrown (which _doesn't_ inherit from `Exception`).

gsnedders commented on Tools/Scripts/webkitpy/layout_tests/servers/web_platform_test_server.py:
```diff
@@ -180,6 +185,9 @@ def _spawn_process(self):
         self._filesystem.write_text_file(self._pid_file, str(self._process.pid))
 
+        if self._dns_server:
+            self._dns_server.start_thread()
```
I think @JonWBedard's comment above is probably more applicable here, and specifically whether we want this to be a subprocess or a thread.

The main downside to this being a thread is the fact that while the DNS server is responding its thread needs to hold the [GIL](https://docs.python.org/3/glossary.html#term-global-interpreter-lock), and potentially will have had wait to acquire the GIL.

That said, if this didn't have any notable impact to EWS runtime then it's probably fine.

sysrqb committed:
Add support for running a local DNS resolver for WPT
https://bugs.webkit.org/show_bug.cgi?id=261038
rdar://82927182

Reviewed by NOBODY (OOPS!).

Some ports don't have support for resolving arbitrary domains, but
web-platform-tests rely on this capability. Currently, the Apple ports only
use localhost, and 127.0.0.1 as a workaround where it's necessary, but this
affects our test coverage. This patch adds a very basic DNS resolver and
introduces a new command-line argument for run-webkit-tests; this is disabled
by default.

In 254273@main, glib ports gained support for resolving arbitrary domains,
and this moves Apple ports closer to supporting that. We expect to enable
this by default, along with rebaselining, in a later commit.

Previous attempts were made in https://bugs.webkit.org/show_bug.cgi?id=245294,
and 264076@main introduced part of that behavior for a different reason. In
some situations, name resolutions falls back on that SPI, as well.

The basic DNS resolver only resolves A records for names ending with
"localhost" and for the host names defines by the port's localhost_aliases.
The server always returns 127.0.0.1 as the answer for those hosts. All other
queries return NXDomain. This resolver depends on the dnslib Python library.

On Apple ports, the iOS Simulator and Mac use different implementations. The
iOS simulator part is in the Network process because we need to define the
resolver in-process otherwise it won't be used. The way the resolver is
configured on Mac isn't supported on the simulator. On Mac, we use SPI and a
Network Extensions policy so this works when a VPN is enabled. We do this in
WebKitTestRunner so we don't need to add additional entitlements and weaken
the sandbox for the network process.

Commit:
fixup! Enable the feature for testing

* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(parse_args):
